engine/src/main/java/org/hibernate/validator/internal/util/privilegedactions/GetValueExtractorsFromServiceLoader.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.util.privilegedactions;
+
+import java.security.PrivilegedAction;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ServiceConfigurationError;
+import java.util.ServiceLoader;
+
+import javax.validation.valueextraction.ValueExtractor;
+
+/**
+ * @author Guillaume Smet
+ */
+public class GetValueExtractorsFromServiceLoader implements PrivilegedAction<List<ValueExtractor<?>>> {
```

Comment:
This looks quite the same as `GetConstraintValidatorsFromServiceLoader`. Couldn't it be one class taking the type to load as a parameter?